GitUI/HelperDialogs/FormResetCurrentBranch.cs

```diff
@@ -31,7 +31,7 @@ private FormResetCurrentBranch()
         public FormResetCurrentBranch(GitUICommands commands, GitRevision revision, ResetType resetType = ResetType.Mixed)
             : base(commands)
         {
-            Revision = revision;
+            Revision = revision ?? throw new NotSupportedException(Strings.NoRevision);
```

Comment:
Throwing in the ctor prevents the object from being disposed

Reply:
We could change back to throwing in the load again, but you get the NBug anyway?

Reply:
The class must not be used without a revision.

Reply:
Throwing in the load method seems a good compromise. Is everyone OK?

Reply:
> Throwing in the load method seems a good compromise. Is everyone OK?

Yes, change back to that, it is slightly better as a template and if you dismiss the NBug
Whoever doing it, squash and rebase as well so it is ready for 3.4

Reply:
I won't block this. Though you cannot convince me that this was a clear approach. It is not worth having a further discussion on the philosophy of C# regarding this.
But the `null` check could be done by a factory method, which then calls the private ctor.

Reply:
done -- not squashed and rebased yet for easier review